worlds/ahit/Regions.py

```diff
@@ -0,0 +1,947 @@
+        green = create_region_and_connect(w, "Green Clean Station", "-> Green Clean Station", metro_freeroam)
+        pink = create_region_and_connect(w, "Pink Paw Station", "-> Pink Paw Station", metro_freeroam)
+        create_region_and_connect(w, "Bluefin Tunnel", "-> Bluefin Tunnel", metro_freeroam)  # No manhole
+
+        create_region_and_connect(w, "Yellow Overpass Manhole", "-> Yellow Overpass Manhole", yellow)
+        create_region_and_connect(w, "Green Clean Manhole", "-> Green Clean Manhole", green)
+        create_region_and_connect(w, "Pink Paw Manhole", "-> Pink Paw Manhole", pink)
+
+        create_rift_connections(w, create_region(w, "Time Rift - Rumbi Factory"))
+        create_thug_shops(w)
+
+
+def create_rift_connections(world: "HatInTimeWorld", region: Region):
+    i = 1
+    for name in rift_access_regions[region.name]:
+        act_region = world.multiworld.get_region(name, world.player)
+        entrance_name = f"{region.name} Portal - Entrance {i}"
+        connect_regions(act_region, region, entrance_name, world.player)
+        i += 1
+
+    # fix for some weird keyerror from tests
+    if region.name == "Time Rift - Rumbi Factory":
+        for entrance in region.entrances:
+            world.multiworld.get_entrance(entrance.name, world.player)
+
+
+def create_tasksanity_locations(world: "HatInTimeWorld"):
+    ship_shape: Region = world.multiworld.get_region("Ship Shape", world.player)
+    id_start: int = TASKSANITY_START_ID
+    for i in range(world.options.TasksanityCheckCount.value):
+        location = HatInTimeLocation(world.player, f"Tasksanity Check {i+1}", id_start+i, ship_shape)
+        ship_shape.locations.append(location)
+
+
+def is_valid_plando(world: "HatInTimeWorld", region: str, is_candidate: bool = False) -> bool:
+    # Duplicated keys will throw an exception for us, but we still need to check for duplicated values
+    if is_candidate:
+        found_list: List = []
+        old_region = region
+        for name in world.options.ActPlando.keys():
+            act = world.options.ActPlando.get(name)
+            if act == old_region:
+                region = name
+                found_list.append(name)
+
+        if len(found_list) == 0:
+            return False
+
+        if len(found_list) > 1:
+            raise Exception(f"ActPlando ({world.multiworld.get_player_name(world.player)}) - "
+                  f"Duplicated act plando mapping found for act: \"{old_region}\"")
+    elif region not in world.options.ActPlando.keys():
+        return False
+
+    if region in blacklisted_acts.values() or (region not in act_entrances.keys() and "Time Rift" not in region):
+        return False
+
+    act = world.options.ActPlando.get(region)
+    try:
+        world.multiworld.get_region(region, world.player)
+        world.multiworld.get_region(act, world.player)
+    except KeyError:
+        return False
```

Comment:
This is really just to make sure an exception isn't thrown for when things like DLC levels are put in the plando section. I'd rather just print a warning for that instead of having it throw an exception, in case the player decides to randomize their DLC options or something.